src/Swarm/Game/Step/Const.hs

```diff
@@ -422,6 +423,9 @@ execConst runChildProg c vs s k = do
         -- take recipe inputs from inventory and add outputs after recipeTime
         robotInventory .= invTaken
         traverse_ (updateDiscoveredEntities . snd) (recipe ^. recipeOutputs)
+        -- Grant CraftedBitcoin achievement
+        when (name == "bitcoin" && not creative && not sys) $ grantAchievement CraftedBitcoin
```

Comment:
I think I would like to make `ValidInCreativeMode` and `ValidForSystemRobot` into properties of the achievements themselves.  Then the `grantAchievement` function would perform the verification (perhaps with the addition of a `HasRobotStepState` constraint).